trio/_ssl.py

```diff
@@ -431,7 +442,13 @@ def _check_status(self):
     # comments, though, just make sure to think carefully if you ever have to
     # touch it. The big comment at the top of this file will help explain
     # too.
-    async def _retry(self, fn, *args, ignore_want_read=False, is_handshake=False):
+    async def _retry(
+        self,
+        fn: Callable[..., T],
+        *args: Any,
```

Comment:
`object` probably works here

Reply:
To be honest I am confused why `Any` would be any worse

Reply:
If we ever accidentally use an `Any` value, then that's really un-type-safe. Also, inference and everything can lead to badly inferred types with `Any` (people normally don't want that inferred).

Reply:
Changed in acfcead1

Reply:
https://typing.readthedocs.io/en/latest/source/best_practices.html#using-any-and-object
https://adamj.eu/tech/2021/05/07/python-type-hints-use-object-instead-of-any/

it took me a bit to grok the difference, but A5rocks has shown me the light